### ISSUE TITLE
Regarding the use of MLflow autologging feature (Phenomenon where two Experiments are created)

### ISSUE THREAD

**Issue by Cow-Kite** [ultralytics/ultralytics#22921]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am currently logging the YOLO training process to MLflow.

In this process, I created the experiment name using an environment variable (EXPERIMENT_NAME: yolo_diamond_detection_v1).

I only want to see the yolo_diamond_detection_v1 experiment in the MLflow web UI.

However, in reality, an experiment named /home/jovyan/work/runs/yolo_diamond_detection_v1 is also being created. (It only creates the experiment, but no run information is logged to it.)

<img width="1198" height="121" alt="Image" src="https://github.com/user-attachments/assets/f40d0a82-a395-47f7-bac2-e053300fc89b" />

Could you tell me the cause of this issue and how to resolve it?

I am attaching the relevant code. Thank you.

```py
import os
import mlflow
from datetime import datetime
from modules.yolo_loader import YOLOLoader
from ultralytics import YOLO

MLFLOW_TRACKING_URI = os.getenv('MLFLOW_TRACKING_URI')

DATASET = os.getenv('DATASET')
LABEL_STORAGE = os.getenv('LABEL_STORAGE')
IMAGE_STORAGE = os.getenv('IMAGE_STORAGE')
CLASS_NAMES = os.getenv('CLASS_NAMES')

BASE_MODEL = os.getenv('BASE_MODEL')
EPOCHS = int(os.getenv('EPOCHS','100'))
BATCH_SIZE = int(os.getenv('BATCH_SIZE','16'))
LEARNING_RATE = float(os.getenv('LEARNING_RATE','0.01'))
IMG_SIZE = int(os.getenv('IMG_SIZE','320'))

EXPERIMENT_NAME = os.getenv('EXPERIMENT_NAME')
RUN_NAME = os.getenv('RUN_NAME')

if __name__ == "__main__":

    if not DATASET: 
        from label2yolo import *
        LABEL_PATH = fine_export_json_file(LABEL_STORAGE)
        IMAGE_PATH = os.path.join('/home/jovyan/work/storages/',IMAGE_STORAGE)
        DATASET = os.path.join('/home/jovyan/work/yolo/',run_yolo_converter(LABEL_PATH,CLASS_NAMES,IMAGE_PATH))

    mlflow.set_tracking_uri(MLFLOW_TRACKING_URI)
    mlflow.set_experiment(EXPERIMENT_NAME)
        
    print("=" * 50)
    print("YOLO 학습 시작")
    print(f"모델: {BASE_MODEL}")
    print(f"데이터셋: {DATASET}")
    print(f"에포크: {EPOCHS}")
    print(f"Experiment: {EXPERIMENT_NAME}")
    print("=" * 50)

    if not RUN_NAME:
        timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
        run_name = f"yolo_{timestamp}"
    else:
        run_name = RUN_NAME

    project_path = os.path.join(str('/home/jovyan/work/runs/'),EXPERIMENT_NAME)

    with mlflow.start_run():
        # 파라미터 로깅
        mlflow.log_param("model", BASE_MODEL)
        mlflow.log_param("dataset", DATASET)
        mlflow.log_param("epochs", EPOCHS)
    
        # 모델 로드
        loader = YOLOLoader(yolo_model=BASE_MODEL)
        model = loader.load()
        
        # 학습 
        results = model.train(
            data = DATASET,
            epochs = EPOCHS,
            batch = BATCH_SIZE,
            lr0 = LEARNING_RATE,
            imgsz = IMG_SIZE,
            project = project_path,
            name = run_name,
            exist_ok = True,
            mlflow=False
        )
                
        # 학습된 모델 저장
        best_model_path = results.save_dir / 'weights' / 'best.pt'
        print('학습 완료!')
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Cow-Kite, thank you for your detailed question and for using Ultralytics with MLflow 🚀  
This is an automated response to help you get support as quickly as possible — an Ultralytics engineer will review your issue and follow up with you soon 🙂.

In the meantime, we recommend checking the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. A lot of common integration and logging behaviors are covered there and may help narrow down what you are seeing with MLflow autologging and experiment naming.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (as in your case with MLflow and YOLO training integration), please:
-Ensure you’ve included:
- the exact training command or script you’re running
- relevant environment details (Python version, OS, `ultralytics` version, MLflow version)
- any logs or console output related to experiment creation
-Confirm you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), as these can affect how and where runs are saved and organized.

You’re also very welcome to share more context (for example, whether you are using notebooks, containers, or a managed platform), which will help the engineers better understand why multiple experiments appear in the MLflow UI.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Before an engineer investigates further, please ensure you have upgraded to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a **Python>=3.8** environment with **PyTorch>=1.8** to verify the behavior still occurs on the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

If you can reproduce the dual-experiment behavior in one of these environments and share the exact steps, that will greatly help the team understand what’s happening.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You can change the environment variable `EXPERIMENT_NAME` to `MLFLOW_EXPERIMENT_NAME` and `RUN_NAME` to `MLFLOW_RUN`.

Ultralytics performs MLFlow logging automatically. You shouldn't be starting the logging manually
